[DG-573] Add description field to entities/relationships (#7600)
segmentio/segment-docs: src/unify/data-graph/index.md

```

### 3a: Define entities

The first step in creating a Data Graph is to define your entities. An entity corresponds to a table in the warehouse.

| Parameters                                  | Definition                                                                                                                                                                                               |
| ------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `entity`                                    | An immutable slug for the entity, and will be treated as a delete if you make changes. The slug must be in all lowercase, and supports dashes or underscores (e.g `account-entity` or `account_entity`). |

Reviewer: ```suggestion
| `entity`                                    | An immutable slug for the entity, and will be treated as a delete if you make changes. The slug must be in all lowercase, and supports dashes or underscores (for example, `account-entity` or `account_entity`). |
```

> | `description` (*Optional*)                  | An optional descriptor used to add additional context to the entity (E.g. table grain, cadence at which the table/view is refreshed, etc).                                                                                                                                        |

Reviewer: ```suggestion
| `description` (*Optional*)                  | An optional descriptor used to add additional context to the entity (for example, table grain, cadence at which the table/view is refreshed).                                                                                                                                        |
```

> | `name`                                      | A label displayed throughout your Segment space for Linked Events, Linked Audiences, etc. This name can be modified at any time.                                                                         |

Reviewer: ```suggestion
| `name`                                      | A label displayed throughout your Segment space for Linked Events, Linked Audiences. This name can be modified at any time.                                                                         |
```

- **[1:many relationship](#define-a-1many-relationship):** For example, an `account` can have many `carts`, but each `cart` can only be associated with one `account`.
- **[many:many relationship](#define-manymany-relationship):** For example, a user can have many `carts`, and each `cart` can have many `products`. However, these `products` can also belong to many `carts`.

#### Define profile-to-entity relationship

This is the first level of relationships and a unique type of relationship between the Segment profile entity and a related entity.

| Parameters                 | Definition                                                                                                                                                                                                 |
| -------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `relationship`             | An immutable slug for the relationship, and will be treated as a delete if you make changes. The slug must be in all lowercase, and supports dashes or underscores (like `user-account` or `user_account`) |
> | `name`                     | A label displayed throughout your Segment space for Linked Events, Linked Audiences, etc. This name can be modified at any time                                                                            |
> | `description` (*Optional*) | An optional descriptor used to add additional context to the relationship.                                                                                                                                 |

Reviewer: ```suggestion
| `relationship`             | An immutable slug for the relationship, and will be treated as a delete if you make changes. The slug must be in all lowercase, and supports dashes or underscores (like `user-account` or `user_account`). |
| `name`                     | A label displayed throughout your Segment space for Linked Events, Linked Audiences. This name can be modified at any time.                                                                            |
| `description` (*Optional*) | An optional descriptor used to add additional context to the relationship.                                                                                                                                 |
```

> | `related_entity`           | References your already defined entity                                                                                                                                                                     |

Reviewer: ```suggestion
| `related_entity`           | This references the entity that's already defined.                                                                                                                                                                     |
```

```

#### Define a 1:many relationship

For 1:many relationships, define the join on between the two entity tables using the spec below.

| Parameters                 | Definition                                                                                                                                                                                                                                         |
| -------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `relationship`             | An immutable slug for the relationship, and will be treated as a delete if you make changes. The slug must be in all lowercase, and supports dashes or underscores (like `user-account` or `user_account`)                                         |
> | `name`                     | A label displayed throughout your Segment space for Linked Events, Linked Audiences, and so on. This name can be modified at any time                                                                                                              |
> | `description` (*Optional*) | An optional descriptor used to add additional context to the relationship.                                                                                                                                                                         |
> | `related_entity`           | References your already defined entity                                                                                                                                                                                                             |
> | `join_on`                  | Defines relationship between the two entity tables `[lefty entity slug].[column name] = [right entity slug].[column name]`. Note that since you’re referencing the entity slug for the join on, you do not need to define the full table reference |

Reviewer: ```suggestion
| `relationship`             | An immutable slug for the relationship, and will be treated as a delete if you make changes. The slug must be in all lowercase, and supports dashes or underscores (like `user-account` or `user_account`).                                         |
| `name`                     | A label displayed throughout your Segment space for Linked Events, Linked Audiences, and so on. This name can be modified at any time                                                                                                              |
| `description` (*Optional*) | An optional descriptor used to add additional context to the relationship.                                                                                                                                                                         |
| `related_entity`           | References your already defined entity                                                                                                                                                                                                             |
| `join_on`                  | Defines relationship between the two entity tables `[lefty entity slug].[column name] = [right entity slug].[column name]`. Note that since you’re referencing the entity slug for the join on, you do not need to define the full table reference. |
```

```

#### Define many:many relationship

For many:many relationships, define the join on between the two entity tables with the `junction_table`.

> warning ""
> Attributes from a junction table are not referenceable via the Linked Audience builder. If a marketer would like to filter upon a column on the junction table, you must define the junction as an entity and define a relationship.

| Parameters               | Definition                                                                                                                                                                                                 |

> | ------------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
> | `relationship`           | An immutable slug for the relationship, and will be treated as a delete if you make changes. The slug must be in all lowercase, and supports dashes or underscores (like `user-account` or `user_account`) |
> | `name`                   | A label displayed throughout your Segment space for Linked Events, Linked Audiences, and so on. This name can be modified at any time                                                                      |
> | (Optional) `description` | An optional descriptor used to add additional context to the relationship                                                                                                                                  |
> | `related_entity`         | References your already defined entity                                                                                                                                                                     |

Reviewer: ```suggestion
| ------------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `relationship`           | An immutable slug for the relationship, and will be treated as a delete if you make changes. The slug must be in all lowercase, and supports dashes or underscores (like `user-account` or `user_account`). |
| `name`                   | A label displayed throughout your Segment space for Linked Events, Linked Audiences, and so on. This name can be modified at any time.                                                                      |
| (Optional) `description` | An optional descriptor used to add additional context to the relationship.                                                                                                                                  |
| `related_entity`         | This references your defined entity.                                                                                                                                                                     |
```

**Junction table spec**

| Parameters      | Definition                                                                                                                                                                                                                                                             |
| --------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `table_ref`     | Defines the fully qualified table reference to the join table: `[database name].[schema name].[table name]`. Segment flexibly supports tables, views and materialized views                                                                                            |
> | `primary_key`   | The unique identifier for the given table. Must be a column with unique values per row                                                                                                                                                                                 |
> | `left_join_on`  | Define the relationship between the left entity table and the junction table: `[left entity slug].[column name] = [junction table column name]`. Note that schema and table are implied within the junction table column name, so you do not need to define it again   |
> | `right_join_on` | Define the relationship between the junction table and the right entity table: `[junction table column name] = [right entity slug].[column name]`. Note that schema and table are implied within the junction table column name, so you do not need to define it again |

Reviewer: ```suggestion
| `table_ref`     | Defines the fully qualified table reference to the join table: `[database name].[schema name].[table name]`. Segment flexibly supports tables, views and materialized views.                                                                                           |
| `primary_key`   | The unique identifier for the given table. Must be a column with unique values per row.                                                                                                                                                                                 |
| `left_join_on`  | Defines the relationship between the left entity table and the junction table: `[left entity slug].[column name] = [junction table column name]`. Note that schema and table are implied within the junction table column name, so you do not need to define it again.   |
| `right_join_on` | Define the relationship between the junction table and the right entity table: `[junction table column name] = [right entity slug].[column name]`. Note that schema and table are implied within the junction table column name, so you do not need to define it again. |
```
